recipes/codan/build.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+
+mkdir -p $PREFIX/bin
+chmod +x bin/*
+cp bin/* $PREFIX/bin
+
+mkdir -p $PREFIX/for_MacOS_users
+chmod +x for_MacOS_users/*
+cp for_MacOS_users/* $PREFIX/for_MacOS_users
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+  chmod +x for_MacOS_users/*
+  cp for_MacOS_users/* $PREFIX/bin
+fi
```

Comment:
Since this is compiled, it might make since instead to keep this as a non-generic package and `mv` the mac binary into bin/ in the build script. This then means only the appropriate binaries are shipped with each of the OSX and Linux packages. You can then shorten post-link.sh

Reply:
Unfortunately (and I don't know the reason), the installation process does not allow these lines (from 11 to 14) to properly work in the build.sh. That's why we had to make a directory in the PREFIX to keep the "for_MacOS_users" in the installation path and, then, use the post-link.sh to cp the binary to the final bin folder (we did several tests to finally reach into this working build). I decided to keep these lines also in the build.sh, in the case conda get any release that allows it to work in the build.sh. I think it is better to have it copied twice and CodAn works in MacOS than not copying and CodAn fails in MacOS.
Let me know your thoughts about it... Should we remove these lines from the build.sh?

Reply:
It was labeled as a generic package, so the script was never run on OSX. Generic packages are built on Linux for all architectures.

Reply:
It may be the reason behind the failure when keeping these commands in the build.sh only. Is there a better way to set this? Should we try to change the noarch option (the Conda documentation is not good enough to help design it in a better way).
In our tests, using these commands in the "post-link.sh" makes it properly work in OSX, despite being build in a Linux architecture.

Reply:
Yeah, the best way is to remove the noarch stuff. The linter will complain, so you can either put the compilers in the build section or surpress the linting error https://bioconda.github.io/contributor/linting.html . Then there will be a separate package for OSX and the `cp` stuff in the post-link script can be removed.